app/backend/src/clj/teet/task/task_tx.clj

```diff
@@ -0,0 +1,19 @@
+(ns teet.task.task-tx
+  "Transaction functions for tasks."
+  (:require [teet.task.task-db :as task-db]
+            [teet.meta.meta-model :as meta-model]
+            [datomic.ion :as ion]))
+
+(defn delete-task
+  "Check all preconditions for deleting a task. If precondition
+  fails, cancel the transactions. Otherwise return a tx that marks
+  the task as deleted."
+  [db user task-id]
+  (cond
+    (task-db/task-has-files? db task-id)
```

Comment:
> Manager can see and use the Delete button on Edit task modal only when the task is not marked as "Separate contract (in THK)" and task does not have any current (undeleted files) files. 

Should this AC be checked here also?

Reply:
Checked in pre condition of the command, as all authorization stuff is checked at command level, not db